components/resource_metering/src/reporter.rs

```diff
@@ -110,13 +114,13 @@ impl Runnable for ResourceMeteringReporter {
                 if !new_config.should_report() {
                     self.client.take();
                     self.cpu_records_collector.take();
+                    self.config = new_config;
```

Comment:
why moving the outer to here?

Reply:
since the `if` and `else` branch both need this, but the order is different, so I move to here.